checkita-core/src/main/scala/ru/raiffeisen/checkita/connections/kafka/KafkaConnection.scala

```diff
@@ -107,7 +104,12 @@ case class KafkaConnection(config: KafkaConnectionConfig) extends DQConnection w
       case KafkaTopicFormat.Json =>
         from_json(binaryColumn.cast(StringType), schemaGetter("JSON").schema).alias(colName)
       case KafkaTopicFormat.Xml =>
-        from_json(xmlToJson(binaryColumn.cast(StringType)), schemaGetter("XML").schema).alias(colName)
+        val xmlSchema = schemaGetter("XML").schema
+        from_json(
+          to_json(
+            from_xml(binaryColumn.cast(StringType), xmlSchema)
```

Comment:
Can't we just use `from_xml` directly without converting it to json and then parsing the json string? Previously we needed `from_json` since xml string was converted to json and then json string was parsed to dataframe columns. It looks like from_xml is able to parse raw xml directly to dataframe columns.

Reply:
Yes, we can do like that. Good point